test/logger.test.js

```diff
@@ -280,8 +280,19 @@ test('should throw in case external logger does not has a child property', t =>
   try {
     Fastify({ logger: loggerInstance })
   } catch (err) {
-    t.equal(err.code, 'FST_ERR_LOG_INVALID_OBJECT')
-    t.equal(err.message, "Invalid logger object provided. The logger instance should have a 'child' function. child")
+    t.equal(err.code, 'FST_ERR_LOG_INVALID_LOGGER')
+    t.equal(err.message, "Invalid logger object provided. The logger instance should have these functions(s): 'child'.")
+  }
+})
+
+test('should throw in case a partially matching logger is provided', t => {
+  t.plan(2)
+
+  try {
```

Comment:
It is better to use t.throws method.

Reply:
I think this `try/catch` is acceptable. The error is being inspected for more than "did it throw".

Reply:
As second parameter t.throws can have the expected Error. 

But i am ok, as long as we habe t.plan set.